### ISSUE TITLE
mutable struct arguments aren't auto dereferenced inside function 

### ISSUE THREAD

**Issue by SleepyRoy** [vlang/v#8560]
**V version:** 
master

**OS:** 
win10

**What did you do?**
```V
struct St {
mut:
    n int
}

fn f(mut x St) {
    mut y := St{n: 2}
    a := x
    b := y
    x.n = 3
    y.n = 4
    println('$a.n $b.n')
}

fn main() {
    mut x := St{ n: 1 }
    f(mut x)
}
```

**What did you expect to see?**
`1 2`
 
**What did you see instead?**
`3 2`

**Analysis**
Mutable arguments are always passed by reference and inside function they're treated as references (not auto derefed when it's used), so in above example `a := x` actually gives a reference instead of a value. But as discussed in #8548 and agreed by Alex, mutable arguments should always be auto derefed if needed (so here `a` should be a value, not a ref), making behavior consistent (with immutable arguments) in V.

**Comment by SleepyRoy** [comment]
A reduced example:
```V
struct Foo {
mut:
	foo int
}

fn main() {
	mut f := Foo{foo: 1}
	foo(mut f)
}

fn foo(mut f Foo) {
	f2 := f
	f.foo = 100
	println(f)
	println(f2)
}
```

got
```
Foo{
    foo: 100
}
&Foo{
    foo: 100
}
```

expected
```
Foo{
    foo: 100
}
Foo{
    foo: 1
}
```

**Comment by SleepyRoy** [comment]
This is fixed by #8745.
